stingray/lightcurve.py

```diff
@@ -595,12 +595,34 @@ def rebin(self, dt_new=None, f=None, method='sum'):
             raise ValueError("New time resolution must be larger than "
                              "old time resolution!")
 
-        bin_time, bin_counts, bin_err, _ = \
-            utils.rebin_data(self.time, self.counts, dt_new,
-                             yerr=self.counts_err, method=method)
+
+        if self.gti is None:
+
+            bin_time, bin_counts, bin_err, _ = \
+                utils.rebin_data(self.time, self.counts, dt_new,
+                                 yerr=self.counts_err, method=method)
+
+        else:
+            bin_time, bin_counts, bin_err = [], [], []
+            for g  in self.gti:
+                # find start and end of GTI segment in data
+                start_ind = self.time.searchsorted(g[0])
+                end_ind = self.time.searchsorted(g[1])
+
+                t_temp = self.time[start_ind:end_ind]
+                c_temp = self.counts[start_ind:end_ind]
+                e_temp = self.counts_err[start_ind:end_ind]
+
+                bin_t, bin_c, bin_e, _ = \
+                    utils.rebin_data(t_temp, c_temp, dt_new,
+                                     yerr=e_temp, method=method)
+
+                bin_time.extend(bin_t)
+                bin_counts.extend(bin_c)
+                bin_err.extend(bin_e)
 
         lc_new = Lightcurve(bin_time, bin_counts, err=bin_err,
-                            mjdref=self.mjdref, dt=dt_new)
+                            mjdref=self.mjdref, dt=dt_new, gti=self.gti)
```

Comment:
To a first glance, it would seem that the issue was just here in this line. Have you tried to only change this and not the rest?

Reply:
Here's what happens when I do what you suggest.

It still doesn't look right, I suspect because of the way the `rebin` method counts bins.

![herx1_rebin_test1](https://user-images.githubusercontent.com/5237391/33140354-8e0ed3fe-cf64-11e7-9b2f-a959bb35c8fb.png)